### PR TITLE
Docs Cleanup for QuickStart and Credentials

### PR DESCRIPTION
**Pull request by randybias**
Standardized and streamlined quick start guides, glossaries, and setup instructions for Project 0x2A

**Details:**

- Enhanced README with standardized formatting for consistency.
- Updated glossary to include new terminology, reorganized definitions for clarity.
- Improved quick start guides for AWS, Azure, and vSphere:
  - Added consistent heading structure and note formats.
  - Introduced explicit template retrieval steps and validation checks.
  - Revised syntax for clarity in command examples and YAML configurations.
- Enhanced MkDocs configuration:
  - Added support for tables, admonition, and other markdown extensions.
  - Updated navigation structure for improved user experience.

### CODE REVIEW

**Review by randybias — COMMENTED** [review]
Looks good.... Let's see if this will auto-merge.

File: docs/quick-start/aws.md [hunk 1]
```diff
@@ -156,7 +174,7 @@ metadata:
   name: my-aws-managedcluster1
   namespace: hmc-system
 spec:
-  template: aws-standalone-cp-0-0-2
+  template: <Template Name> # The name of the template you want to use from above
```
Comment (Schnitzel) [inline docs/quick-start/aws.md]: I don't like this, this is a quick start which should bring the customer to a result with the least amount of decisions as possible. Trying to bring the concept of ClusterTemplates in here is too early. Yes we need a way to explain them how to use templates, how to find them, how to choose them, etc. But in this step of the customer journey I would not have them worry about templates yet.

File: docs/quick-start/aws.md [hunk 2]
```diff
@@ -2,29 +2,29 @@
...
+### Configure AWS IAM
 
 Before launching a cluster on AWS, you need to set up your AWS infrastructure
 with the necessary IAM policies and service account.
 
-> NOTE:
-> Skip steps below if you've already configured IAM policy for your AWS account
+!!! note
+
+    Skip steps below if you've already configured IAM policy for your AWS account
```
Comment (a13x5) [inline docs/quick-start/aws.md]: Is there a specific reason to switch to [admonitions](https://python-markdown.github.io/extensions/admonition/) from [markdown-callouts](https://oprypin.github.io/markdown-callouts/index.html)?

Most of the docs using latter, since it was introduced in #9. I don't see reason to use 2 different plugins with the same functionality, we should choose one.

File: docs/quick-start/aws.md [hunk 3]
```diff
@@ -175,6 +198,5 @@ kubectl -n hmc-system get managedcluster.hmc.mirantis.com my-aws-managedcluster1
 After the cluster is `Ready` you can access it via the kubeconfig, like this:
 
 ```bash
-kubectl -n hmc-system get secret my-aws-managedcluster1-kubeconfig -o jsonpath='{.data.value}' | base64 -d > my-aws-managedcluster1-kubeconfig.kubeconfig
-KUBECONFIG="my-aws-managedcluster1-kubeconfig.kubeconfig" kubectl get pods -A
-```
+( kubectl -n hmc-system get secret my-aws-managedcluster1-kubeconfig -o jsonpath='{.data.value}' | base64 -d > my-aws-managedcluster1-kubeconfig.kubeconfig ) && KUBECONFIG="my-aws-managedcluster1-kubeconfig.kubeconfig" kubectl get pods -A
```
Comment (Schnitzel) [inline docs/quick-start/aws.md]: I don't think this is needed a single oneliner. Having it in two lines explains the customer much better what is going on, and how they can run future `kubectl` commands themselves.

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: docs/quick-start/aws.md (hunk 3 above)
Comment (a13x5) [inline docs/quick-start/aws.md]: NIT:  Use of parenthesis isn't needed in this case and it's a bit unconventional outside of conditional statement.

**Review by a13x5 — CHANGES_REQUESTED** [review]
Don't understand the introduction of admonitions in the quick start guides.
2A -> 0x2A rename probably should be discussed with @Schnitzel 

Rest looks good

File: docs/quick-start/aws.md (hunk 2 above)
Comment (randybias) [reply docs/quick-start/aws.md]: I went with the default in the Material theme.

File: docs/quick-start/aws.md (hunk 1 above)
Comment (randybias) [reply docs/quick-start/aws.md]: This was one of the few places I stumbled in the documentation, since the version of the templates, for example, aren't always being updated in the documentation to match what is in a given release.  If we don't do this then every time we version the templates in a release we will have to manually update the quick starts to include any of the template version changes.

I found it more confusing that I didn't know how to figure out why my deployment didn't work than understanding the templates themselves.

File: docs/quick-start/aws.md (hunk 3 above)
Comment (randybias) [reply docs/quick-start/aws.md]: It makes sure everything in the sub shell exits cleanly in order to proceed.  If it's unconventional then I've been doing it unconventionally since the 1980s.  Regardless, due to Michael's other feedback this is split back into two commands so it's moot.

**Comment by randybias** [comment]
> Don't understand the introduction of admonitions in the quick start guides. 2A -> 0x2A rename probably should be discussed with @Schnitzel
> 
> Rest looks good

For whatever it's worth, I was changing it everywhere.  Again, it was the default in the Material theme so I thought it made sense.  I agree that two different mechanisms for this doesn't make sense.  So after y'all make a decision I'll just go whichever way you decide.

If you decide to go with the admonitions, I will make certain it's done across all of the docs.

**Comment by a13x5** [comment]
>  So after y'all make a decision I'll just go whichever way you decide.

Got it! Let's keep using callouts and not admonitions then, since the syntax is better and it's less changes throughout the code (since we have callouts everywhere).

File: docs/quick-start/aws.md (hunk 1 above)
Comment (Schnitzel) [reply docs/quick-start/aws.md]: >  If we don't do this then every time we version the templates in a release we will have to manually update the quick starts to include any of the template version changes.

yea, I'm ok with that, we have to update the docs anyway with every new 2A release. Rather spend some time updating docs and provide a super simple quickstart, than having confused first time users.
Also we can automate the updating of the docs.

File: docs/glossary.md [hunk 4]
```diff
@@ -6,16 +6,35 @@ more common ones that we feel may need a little clarity in the way we use
...
+
+### Managed Cluster
+A Kubernetes cluster created and managed by Project 2A.
+
+### Management Cluster
+The Kubernetes cluster where 2A is installed and from which all other managed clusters will
+be managed from.
+
+### Provider (also Infrastructure Provider)
+<<<<<<< HEAD
```
Comment (Schnitzel) [inline docs/glossary.md]: some merge conflict pieces

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: docs/quick-start/aws.md (hunk 1 above)
Comment (randybias) [reply docs/quick-start/aws.md]: Alright.  I'll revert this.

Comment (randybias) [reply docs/quick-start/aws.md]: Fixed.

File: docs/glossary.md (hunk 4 above)
Comment (randybias) [reply docs/glossary.md]: fixed

Comment (randybias) [reply docs/glossary.md]: Everything converted to callouts.

File: docs/quick-start/aws.md (hunk 2 above)
Comment (randybias) [reply docs/quick-start/aws.md]: Fixed.

File: docs/glossary.md [hunk 5]
```diff
@@ -5,17 +5,32 @@ attempt to clarify some of the unique terms and concepts we use or explain
...
+### Cloud Controller Manager
+Cloud Controller Manager (CCM) is a Kubernetes component that embeds
+logic to manage a specific infrastructure provider.
+
+### ClusterIdentity
+ClusterIdentity is a Kubernetes object that references a Secret object containing for
+a specific infrastructure provider.
+
+### Credential
+A Credential is a Kubernetes object that provides 2A with the necessary credentials to manage
```
Comment (bnallapeta) [inline docs/glossary.md]: This is sort of implying that credential is a generic kubernetes object. Its not. Better to rephrase this as:

Credential is a Custom Resource (CR) in HMC that supplies 2A with the necessary credentials to manage a specific infrastructure.

File: docs/glossary.md [hunk 6]
```diff
@@ -5,17 +5,32 @@ attempt to clarify some of the unique terms and concepts we use or explain
...
+a specific infrastructure. The credential object references other Kubernetes objects with infrastructure specific credentials like access keys, passwords, certificates etc.
+
+### Managed Cluster
+A Kubernetes cluster created and managed by Project 2A.
+
+### Management Cluster
+The Kubernetes cluster where 2A is installed and from which all other managed clusters will
+be managed from.
+
+### Provider (also Infrastructure Provider)
```
Comment (bnallapeta) [inline docs/glossary.md]: I think we should separate the two. 

Provider is a HMC specific concept where as InfrastructureProvider is a Cluster API CRD. 

InfrastructureProvider

InfrastructureProvider is a Kubernetes Custom Resource (CR) that defines the infrastructure-specific configuration needed for managing Kubernetes clusters. It enables Cluster API (CAPI) to provision and manage clusters on a specific infrastructure platform (e.g., AWS, Azure, VMware).

File: docs/index.md [hunk 7]
```diff
@@ -2,48 +2,52 @@
...
 
-Project 2A was created to be a repeatable and secure way to leverage the existing
-Kubernetes ecosystem (e.g. Cluster API) while being able to provide for the range of
-unique use cases that exist within enterprise IT environments.
+## Project 2A vs HMC
 
-## Main Premise
+Project 2A includes all of the components below, but because HMC was the first
+component to be developed, it is still referred to as "HMC" in some documentation.
+In many ways "Project 2A and "HMC" are synonymous.
```
Comment (bnallapeta) [inline docs/index.md]: "Project 2A => "Project 2A"

File: docs/index.md [hunk 8]
```diff
@@ -2,48 +2,52 @@
...
 
-2A is built around the creation of a set of standardised templates that enable
-easy, repeatable cluster deployments and life cycle management.
+2A is built around the creation of a set of standardised templates that enable 
+easy, repeatable cluster deployments and life cycle management. 
 
 The main components of 2A include:
 
- * Hybrid Multi Cluster Management (HMC)
+ * **Hybrid Multi Cluster Management (HMC)**
```
Comment (bnallapeta) [inline docs/index.md]: in the main repo, HMC is expanded to just "Hybrid Multi Cluster" 

https://github.com/Mirantis/hmc?tab=readme-ov-file#mirantis-hybrid-multi-cluster-hmc-codename-project-0x2a

File: docs/index.md [hunk 9]
```diff
@@ -2,48 +2,52 @@
...
 
-    Deployment and lifecycle managment of Kubernetes clusters, including configuration, updates, and other CRUD operations.
+    Deployment and life-cycle management of Kubernetes clusters, including configuration, updates, and other CRUD operations.
 
- * Cluster State Management (SMC)
+ * **Cluster State Management (SMC)**
```
Comment (bnallapeta) [inline docs/index.md]: Is this our own definition? If so, should it not be abbreviated to CSM? 🤔

File: docs/usage/cluster-update.md [hunk 10]
```diff
@@ -5,13 +5,13 @@ To update the `ManagedCluster` update `spec.template` in the `ManagedCluster` ob
 Run:
 
 ```bash
-kubectl patch managedcluster.hmc <cluster-name> -n <cluster-namespace> --patch '{"spec":{"template":"<new-template-name>"}}' --type=merge
+kubectl patch managedcluster.hmc <Your Cluster Name> -n <Namespace> --patch '{"spec":{"template":"<new-template-name>"}}' --type=merge
```
Comment (bnallapeta) [inline docs/usage/cluster-update.md]: standard convention in k8s to use lowercase without spaces.
```
<Your Cluster Name> => <your-cluster-name>
<Namespace> => <namespace>
```

File: docs/usage/cluster-update.md [hunk 11]
```diff
@@ -5,13 +5,13 @@ To update the `ManagedCluster` update `spec.template` in the `ManagedCluster` ob
...
 ```
 
 Then, check the status of the `ManagedCluster` object:
 
 ```bash
-kubectl get managedcluster.hmc <cluster-name> -n <cluster-namespace>
+kubectl get managedcluster.hmc <Your Cluster Name> -n <Namespace>
```
Comment (bnallapeta) [inline docs/usage/cluster-update.md]: same as above

File: docs/usage/create-managed-cluster.md [hunk 12]
```diff
@@ -1,172 +1,233 @@
...
+
+- Create the file with the `ManagedCluster` configuration:
 
     ```yaml
     apiVersion: hmc.mirantis.com/v1alpha1
     kind: ManagedCluster
     metadata:
-      name: <cluster-name>
-      namespace: <cluster-namespace>
+      name: <Your Cluster Name>
```
Comment (bnallapeta) [inline docs/usage/create-managed-cluster.md]: same as above. please fix this in all places.

**Review by bnallapeta — COMMENTED** [review]
few comments to work on.

File: docs/glossary.md (hunk 5 above)
Comment (randybias) [reply docs/glossary.md]: Good catch.

File: docs/glossary.md (hunk 6 above)
Comment (randybias) [reply docs/glossary.md]: Changed provider to CAPI Provider and created new Infrastructure Provider entry.

File: docs/index.md (hunk 8 above)
Comment (randybias) [reply docs/index.md]: We are extremely inconsistent with our naming, acronyms, and more.  I really think we should fix that sooner as the problem just gets worse as we go along.  @Schnitzel @p5ntangle

File: docs/index.md (hunk 9 above)
Comment (randybias) [reply docs/index.md]: I think we were trying to match HMC?  It's inconsistent for sure.  We have three acronyms or abbreviations that are all completely different.

File: docs/usage/cluster-update.md (hunk 10 above)
Comment (randybias) [reply docs/usage/cluster-update.md]: Can you show me this?  I find it harder to read because it looks like other text such as "hmc-system".  I think the variables to be interpolated should be clearly different in some manner.  All caps or something.  As it is every example command is lots of lower case and lots of '-'s and it makes it very challenging to read properly.  Italics.  Anything.  We just need a standard and I don't think lowercase is it. @Schnitzel @p5ntangle

Comment (bnallapeta) [reply docs/usage/cluster-update.md]: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

This states that the resource names must conform to dns subdomain naming convention and thus the no capital letters, no spaces and other such rules. So, we end up following the same convention even for placeholders, but surrounded by <>

If you really want to distinguish that this is a placeholder for the user to fill, then I think its better to do all caps with underscore. 

```
<Your Cluster Name> => <YOUR_CLUSTER_NAME>
<Namespace> => <NAMESPACE>
```

File: docs/index.md (hunk 9 above)
Comment (bnallapeta) [reply docs/index.md]: @p5ntangle ^^

File: docs/usage/cluster-update.md (hunk 10 above)
Comment (randybias) [reply docs/usage/cluster-update.md]: I see... I went through the Cluster API docs.  I still think it's horribly confusing, but I guess if we just want to conform rather than innovate we can do this.

Comment (randybias) [reply docs/usage/cluster-update.md]: OK.  I found this:

https://kubernetes.io/docs/contribute/style/style-guide/

I'll follow that.  I've got to go back in and fix a bunch of things.  Standby.

File: mkdocs.yml [hunk 13]
```diff
@@ -56,7 +56,9 @@ plugins:
 # Page tree
 
 markdown_extensions:
+  - admonition
```
Comment (a13x5) [inline mkdocs.yml]: We should remove it from here, since we decided to use `callouts`

File: docs/usage/cluster-update.md (hunk 10 above)
Comment (randybias) [reply docs/usage/cluster-update.md]: This is all fixed per the style guide.

File: mkdocs.yml (hunk 13 above)
Comment (randybias) [reply mkdocs.yml]: Done.

**Review by a13x5 — APPROVED** [review]
LGTM

File: docs/index.md [hunk 14]
```diff
@@ -3,28 +3,34 @@
...
+component to be developed, it is still referred to as "HMC" in some documentation.
+In many ways "Project 2A" and "HMC" are synonymous.
 
 2A is built around the creation of a set of standardised templates that enable 
 easy, repeatable cluster deployments and life cycle management. 
 
 The main components of 2A include:
 
- * Hybrid Multi Cluster Management (HMC)
+ * **Hybrid Multi Cluster (HMC) Management**
```
Comment (Schnitzel) [inline docs/index.md]:
```suggestion
 * **Hybrid Multi-Cluster Controller (HMC)**
```

Comment (Schnitzel) [reply docs/index.md]: talked to @p5ntangle and we agreed on this naming

File: docs/index.md [hunk 15]
```diff
@@ -3,28 +3,34 @@
...
 The main components of 2A include:
 
- * Hybrid Multi Cluster Management (HMC)
+ * **Hybrid Multi-Cluster Controller (HMC)**
 
-    Deployment and lifecycle managment of Kubernetes clusters, including configuration, updates, and other CRUD operations.
+    Deployment and life-cycle management of Kubernetes clusters, including configuration, updates, and other CRUD operations.
 
- * Cluster State Management (SMC)
+ * **Cluster State Management (SMC)**
```
Comment (Schnitzel) [inline docs/index.md]:
```suggestion
 * **State Management Controller (SMC)**
```

Comment (Schnitzel) [reply docs/index.md]: also verified this with @p5ntangle

**Review by Schnitzel — APPROVED** [review]
